### PR TITLE
fix/kubernetes-pod-status

### PR DESCRIPTION
**Pull request by Nikhil-Singhal-06**
### Description
Pod status action fails to detect the pod status.

### Pull request checklist

- [] [Issue](https://github.com/IntelLabs/scenario_execution/issues) created that explains the change and why it's needed
- [ ] Tests are part of the PR (for bug fixes / features)
- [ ] [Docs](https://intellabs.github.io/scenario_execution/) reviewed and added / updated if needed (for bug fixes / features)
- [ ] Format and Lint checks (`make check`) pass locally
- [ ] PR applies Apache 2.0 License to all code files

### Pull request type

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe)

### CODE REVIEW

File: libs/scenario_execution_kubernetes/scenario_execution_kubernetes/kubernetes_wait_for_pod_status.py [hunk 1]
```diff
@@ -53,13 +53,13 @@ def setup(self, **kwargs):
...
-    def execute(self, target: str, regex: bool, status: tuple, namespace: str):
+    def execute(self, target: str, regex: bool, status: tuple):
         self.target = target
-        self.namespace = namespace
         if not isinstance(status, tuple) or not isinstance(status[0], str):
             raise ValueError("Status expected to be enum.")
         self.expected_status = status[0]
         self.regex = regex
+        self.update_queue = queue.Queue()
         self.current_state = KubernetesWaitForPodStatusState.MONITORING
```
Comment (fred-labs) [inline libs/scenario_execution_kubernetes/scenario_execution_kubernetes/kubernetes_wait_for_pod_status.py:63]: still needed?

**Review by fred-labs — CHANGES_REQUESTED** [review]

File: libs/scenario_execution_kubernetes/scenario_execution_kubernetes/kubernetes_wait_for_pod_status.py [hunk 2]
```diff
@@ -50,35 +42,37 @@ def setup(self, **kwargs):
...
-            self.feedback_message = f"waiting for status of pod '{self.target}'."  # pylint: disable= attribute-defined-outside-init
+            if not self.is_pod:
+                self.feedback_message = f"waiting for status of pod '{self.target}'."  # pylint: disable= attribute-defined-outside-init
             if not self.regex:
                 if item[0] != self.target:
                     continue
             else:
                 if not re.search(self.target, item[0]):
                     continue
+            # self.feedback_message = f"The pod '{self.target}' is currently in '{item[1].lower()}' status."  # pylint: disable= attribute-defined-outside-init
```
Comment (fmirus) [inline libs/scenario_execution_kubernetes/scenario_execution_kubernetes/kubernetes_wait_for_pod_status.py:69]: is this still needed? If not, remove

**Review by fmirus — CHANGES_REQUESTED** [review]

**Review by fmirus — DISMISSED** [review]

File: libs/scenario_execution_kubernetes/scenario_execution_kubernetes/kubernetes_wait_for_pod_status.py [hunk 3]
```diff
@@ -50,35 +42,36 @@ def setup(self, **kwargs):
...
-        self.target = target
         self.namespace = namespace
+        self.target = target
         if not isinstance(status, tuple) or not isinstance(status[0], str):
             raise ValueError("Status expected to be enum.")
         self.expected_status = status[0]
         self.regex = regex
-        self.current_state = KubernetesWaitForPodStatusState.MONITORING
+        self.is_pod = False
+        self.monitoring_thread = threading.Thread(target=self.watch_pods, daemon=True)
```
Comment (fred-labs) [inline libs/scenario_execution_kubernetes/scenario_execution_kubernetes/kubernetes_wait_for_pod_status.py:53]: this creates a new thread for every execution. Instead please rework the previous solution
--> in update() when getting the last element from update_queue (line 58) , store the item in a member variable (e.g. self.current_state). This allows you to keep the state for another execution. Then you can rework the update() function: 
1. move the check within update() into a separate function.
2. update() should look something like that afterwards:

```
if self.update_queue().empty():
  if self.current_state is not None and check(self.current_state):
    return  SUCCESS
  return RUNNING
else:
  while not self.update_queue.empty():
    self.current_state = self.update_queue.get()
    if check(self.current_state):
      return SUCCESS
  return RUNNING
```

File: libs/scenario_execution_kubernetes/scenario_execution_kubernetes/kubernetes_wait_for_pod_status.py [hunk 4]
```diff
@@ -21,27 +21,19 @@
...
         super().__init__()
         self.target = None
-        self.namespace = None
         self.expected_status = None
         self.within_cluster = within_cluster
         self.regex = None
         self.client = None
         self.update_queue = queue.Queue()
-        self.current_state = KubernetesWaitForPodStatusState.IDLE
+        self.is_pod = False
```
Comment (fred-labs) [inline libs/scenario_execution_kubernetes/scenario_execution_kubernetes/kubernetes_wait_for_pod_status.py:36]: please remove (see comment below)

**Review by fred-labs — CHANGES_REQUESTED** [review]

**Review by fred-labs — APPROVED** [review]

**Review by fred-labs — APPROVED** [review]

**Pull request merged**